### ISSUE TITLE
MidiQoL configured keybinds for advantage and disadvantage compatibility

### ISSUE THREAD

**Issue by kgar** [kgar/foundry-vtt-tidy-5e-sheets#240]
Determine what is needed to establish this missing compatibility for MidiQoL, apply any needed changes on the Tidy side, and forward any requests for changes as needed to the Midi QoL gitlab repo.

## Reference

https://discord.com/channels/1167985253072257115/1193391517310660708

Players attempting to use configured keybinds to roll advantage and disadvantage (may also impact critical, versitile and other fast forwarding rolls).

The rolls are not having the advantage or disadvantage applied as expected.

Switching back to core sheets, or old tidy resolves this.

**Comment by kgar** [comment]
Initial diagnosis: 
When "Focus Visible" becomes active on the roll button, then keydown/keyup bindings do not register with Midi.

That is, if I hold the Adv key and press the roll button, I get the intended Adv roll. If I press any other keys that trigger Focus Visible after that (Shift, for example), then Focus Visibility kicks in, and subsequent attempts at handling keydown at the Foundry level are blocked.

**Comment by kgar** [comment]
Focused elements prevent keybind events from going through, as an ingrained feature of Foundry.
I will think about how best to approach this problem.

**Comment by kgar** [comment]
After considering a number of options, this approach will allow me to keep the accessibility features while also forwarding keybind events to the KeyboardManager:
> On keyup and keydown for relevant Tidy buttons, call 
```js
// given ev is a KeyboardEvent
// up is true when a keyup event and false when a keydown event
KeyboardManager.emulateKeypress(
  up, 
  ev.code, 
  {
    altKey: ev.altKey,
    ctrlKey: ev.ctrlKey,
    shiftKey: ev.shiftKey,
    repeat: ev.repeat,
    force: true,
  }
);
```

This particular logic can be centralized and then referenced by all relevant buttons.
Ideally, I should reach for the svelte-y way to do this across the board, in order to make it more obvious.

**Comment by kgar** [comment]
The above approach, and a variety of others I tried after it, have a number of problems.
Core Foundry functionality and how keybindings work prevents a sheet from having both focused button event handling AND keybinding triggers work in tandem.
The only real fix for this is to revert the majority of my buttons to anchor tags.

**Comment by kgar** [comment]
Reference from Tidy community server: https://discord.com/channels/1167985253072257115/1169792539545587733/1196702693637488660

> Midi-QoL leverages the Foundry "Configure Controls" settings to handle Advantage/Disadvantage/some other things. It listens for these key presses while you're on the sheet and clicking a Roll Button, for example.
Meanwhile, when Foundry is deciding whether to trigger the effects of these "Configure Control" keybindings, if it thinks you're focused on an input, like a checkbox or a textbox, it will not trigger keybinding code. For example, if you press Spacebar in a textbox, it doesn't want to toggle Pause/Unpause while you're just trying to type.
As it turns out, because I use buttons for accessibility purposes, Foundry perceives my buttons as inputs that currently have focus, and as a result no Foundry-configured keybindings fire. That means things like Adv/Disadv in Midi-QoL can appear really spotty, not working most of the time, because Foundry thinks I'm focused on an input element and don't want to be bothered... 
I've tried a large number of things, even patching the KeyboardManager ( 😱 ). Each different thing I've tried has problems for one reason or another.
In the end, I don't think Foundry wants for there to be buttons all over my sheet.
I may need to change all the buttons back to anchor <a> tags and the like and just throw that whole section of accessibility goals out the window. Like, no tabbing around to things and being able to hit Spacebar to trigger them.
I figure not many people do this, but for the keyboard-focused people, this was really nice to have. 
Thing is, I wouldn't expect Midi, for example, to try to do something else than what they're doing. The Configure Controls interface and keybindings are part of Foundry. As I now understand it, this is a place in Foundry where I'm just using it in an unsupported way and have to move it closer to the intended HTML makeup.
As always, I don't know nothing. I could be wrong. I'm always keen for any insight from anyone who might be reading.
I will have to emotionally prepare to migrate a LOT of buttons back to anchors.
